kernel/timeout.c

```diff
@@ -21,7 +21,7 @@ static sys_dlist_t timeout_list = SYS_DLIST_STATIC_INIT(&timeout_list);
 
 static struct k_spinlock timeout_lock;
 
-static bool can_wait_forever;
+#define MAX_WAIT (IS_ENABLED(SYS_CLOCK_SLOPPY_IDLE) ? K_FOREVER : INT_MAX)
```

Comment:
should be CONFIG_SYSTEM_CLOCK_SLOPPY_IDLE?

Reply:
Uh... yeah.  :)

Reply:
@andyross still need to fix this one